### PR TITLE
Exp/infersent

### PR DESCRIPTION
**Pull request by sdascoli**

### CODE REVIEW

File: automatic_data_generation/data/base_dataset.py [hunk 1]
```diff
@@ -19,21 +20,22 @@ class BaseDataset(object):
...
-                 none_idx,
-                 input_type,
-                 tokenizer_type,
-                 preprocessing_type,
-                 max_sequence_length,
-                 embedding_type,
-                 embedding_dimension,
-                 max_vocab_size,
-                 output_folder):
+                 dataset_folder:str,
```
Comment (choufractal) [inline automatic_data_generation/data/base_dataset.py]: it's a `Path` actually, not a string

File: automatic_data_generation/data/base_dataset.py [hunk 2]
```diff
@@ -117,7 +119,7 @@ def get_datafields(text, delex, label, intent):
 
     @staticmethod
     @abstractmethod
-    def add_nones(sentences, none_folder, none_idx, none_size):
+    def add_nones(self, sentences, none_folder, none_size=None, none_intents=None, none_idx=None, cosine_threshold=None):
```
Comment (choufractal) [inline automatic_data_generation/data/base_dataset.py]: you need to remove `self` as argument if this method is static

File: automatic_data_generation/data/base_dataset.py [hunk 3]
```diff
@@ -218,6 +222,25 @@ def build_data_files(self, dataset_folder,  dataset_size=None, restrict_intent=N
 
         return new_train_path, new_test_path
 
+    def select_none_intents(self, dataset_folder, restrict_intents, none_folder, cosine_threshold):
+        # select none intents according to overlap with original intents
+        selected_none_intents = []
+        non_intents = []
```
Comment (choufractal) [inline automatic_data_generation/data/base_dataset.py]: unused after that

File: automatic_data_generation/training/losses.py [hunk 4]
```diff
@@ -40,6 +40,6 @@ def compute_bow_loss(batch_size, bow, target):
 def compute_label_loss(logc, target, annealing_strategy, step, k, x, m, none_idx):
     # Negative Log Likelihood
     label_weight = annealing_fn(annealing_strategy, step, k, x, m)
-    nll_label = torch.nn.NLLLoss(reduction='sum', ignore_index=none_idx)
+    nll_label = torch.nn.NLLLoss(reduction='sum')#, ignore_index=none_idx)
```
Comment (choufractal) [inline automatic_data_generation/training/losses.py]: don't you also compute the loss on None intents if you do this?

File: automatic_data_generation/data/base_dataset.py [hunk 5]
```diff
@@ -218,6 +222,25 @@ def build_data_files(self, dataset_folder,  dataset_size=None, restrict_intent=N
...
+        def cosine(u, v):
+            return np.dot(u, v) / (np.linalg.norm(u) * np.linalg.norm(v))
+        intent_vectors = self.load_intent_vectors(dataset_folder)
```
Comment (choufractal) [inline automatic_data_generation/data/base_dataset.py]: don't you need to re-compute the vectors on the restricted dataset? you're kind of cheating if you use the embeddings for the whole one ...

File: automatic_data_generation/evaluation/metrics.py [hunk 6]
```diff
@@ -154,18 +162,17 @@ def intent_classification(candidates, train_path, input_type):
...
-    # try:
     intent_classifier = train_intent_classifier(train_path, input_type)
     for intent, tokenized_sentences in candidates.items():
         sentences = [' '.join(sentence) for sentence in tokenized_sentences]
         preds = intent_classifier.predict(sentences)
-        accuracy = sum([pred==intent for pred in preds]) / len(preds)
+        corrects = [pred == intent for pred in preds]
+        candidates[intent] = [cand for i, cand in enumerate(candidates[intent])
+                              if corrects[i]]
+        accuracy = sum(corrects) / len(preds)
```
Comment (choufractal) [inline automatic_data_generation/evaluation/metrics.py:189]: I took me a while to understand this, you should explicitly return `candidates` as you modify it and it's used by the other metrics. Best would be to return something like `correctly_classified_candidates` and use this as input for BLEU, originality and transfer.

**Review by choufractal — CHANGES_REQUESTED** [review]

**Review by choufractal — APPROVED** [review]
